src/components/tl/ucp/allgather/allgather_linear.c

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+#include "config.h"
+#include "tl_ucp.h"
+#include "allgather.h"
+#include "core/ucc_progress_queue.h"
+#include "tl_ucp_sendrecv.h"
+#include "components/mc/ucc_mc.h"
+
+ucc_status_t ucc_tl_ucp_allgather_linear_start(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_ucp_task_t *task      = ucc_derived_of(coll_task, ucc_tl_ucp_task_t);
+    ucc_tl_ucp_team_t *team      = TASK_TEAM(task);
+    size_t             count     = TASK_ARGS(task).dst.info.count;
+    void              *sbuf      = TASK_ARGS(task).src.info.buffer;
+    void              *rbuf      = TASK_ARGS(task).dst.info.buffer;
+    ucc_memory_type_t  smem      = TASK_ARGS(task).src.info.mem_type;
+    ucc_memory_type_t  rmem      = TASK_ARGS(task).dst.info.mem_type;
+    ucc_datatype_t     dt        = TASK_ARGS(task).dst.info.datatype;
+    ucc_rank_t         trank     = UCC_TL_TEAM_RANK(team);
+    ucc_rank_t         tsize     = UCC_TL_TEAM_SIZE(team);
+    size_t             data_size = (count / tsize) * ucc_dt_size(dt);
+    ucc_status_t       status;
+
+    UCC_TL_UCP_PROFILE_REQUEST_EVENT(coll_task, "ucp_allgather_linear_start",
+                                     0);
+
+    ucc_tl_ucp_task_reset(task, UCC_INPROGRESS);
+
+    /* Copy local data to the receive buffer if not in-place */
+    if (!UCC_IS_INPLACE(TASK_ARGS(task))) {
+        status = ucc_mc_memcpy(PTR_OFFSET(rbuf, data_size * trank), sbuf,
+                               data_size, rmem, smem);
+        if (ucc_unlikely(UCC_OK != status)) {
+            return status;
+        }
+    }
+
+    return ucc_progress_queue_enqueue(UCC_TL_CORE_CTX(team)->pq, &task->super);
+}
+
+/* Get the number of requests in flight to be used for the allgather batched algorithm 
+ * If the number of requests is not specified, use the number of team size - 1
+ * If number of request is bigger than the team size - 1, use the team size - 1
+ */
+static unsigned long get_num_reqs(const ucc_tl_ucp_team_t *team)
+{
+    unsigned long reqs =
+        UCC_TL_UCP_TEAM_LIB(team)->cfg.allgather_batched_num_posts;
+    ucc_rank_t max_req = UCC_TL_TEAM_SIZE(team) - 1;
+    reqs = (reqs > max_req || reqs == UCC_ULUNITS_AUTO) ? max_req : reqs;
+    return reqs;
+}
+
+ucc_status_t ucc_tl_ucp_allgather_linear_init_batched(
```

Comment:
please rename this function now we have ucc_tl_ucp_allgather_linear_init_batched and ucc_tl_ucp_allgather_linear_batched_init and it's not clear which one does what